The temporary variables are not removed on termination

**Describe the bug**
The temporary variables are not removed when the event scope is terminated. They are only removed if the scope is completed.

Temporary variables are used internally to store variables (of jobs or messages) until they are used to apply the variable output mappings of the event scope.   

The variables stays in the state as garbage and contributes to a bigger state :wastebasket: 

**To Reproduce**

The following test case can be added to `ProcessExecutionCleanStateTest` and reproduces the issue.

<details><summary>Unit test</summary>
 <p>

```
@Test
  public void testTerminatingProcessWithServiceTask() {
    // given
    engineRule
        .deployment()
        .withXmlResource(
            Bpmn.createExecutableProcess(PROCESS_ID)
                .startEvent()
                .serviceTask("task", t -> t.zeebeJobType("test").zeebeOutputExpression("x", "y"))
                .endEvent()
                .done())
        .deploy();

    // when
    final var processInstanceKey =
        engineRule.processInstance().ofBpmnProcessId(PROCESS_ID).create();

    engineRule.job().ofInstance(processInstanceKey).withType("test").complete();

    RecordingExporter.incidentRecords(IncidentIntent.CREATED)
        .withProcessInstanceKey(processInstanceKey)
        .await();

    engineRule.processInstance().withInstanceKey(processInstanceKey).cancel();

    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_TERMINATED)
        .withProcessInstanceKey(processInstanceKey)
        .withElementType(BpmnElementType.PROCESS)
        .await();

    // then
    assertThatStateIsEmpty();
  }
```

</p>
</details>

**Expected behavior**
The temporary variables are removed when the event scope is terminated.

**Log/Stacktrace**

```
[Expected all columns to be empty] Expecting empty but was: [TEMPORARY_VARIABLE_STORE]
```

<details><summary>Log from Test Run</summary>
 <p>

```
09:49:47.053 [Broker-0-LogStream-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
09:49:47.321 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Recovering state of partition 1 from snapshot
09:49:47.443 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@1260e741)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@1b7d495b, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@355b7d9b, configuration: Configuration(false)]
09:49:47.614 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
09:49:47.636 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@5ab4a246)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@65993a1d, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@5af938be, configuration: Configuration(false)]
09:49:47.650 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@4a8af8ec)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@780d987e, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@7c91bbdb, configuration: Configuration(false)]
09:49:48.692 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@9f7b9d0)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@37270c2e, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@75fa77e6, configuration: Configuration(false)]
09:49:58.905 [] INFO  io.camunda.zeebe.test - Test failed, following records were exported:
09:49:59.040 [] INFO  io.camunda.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  P[partitionId]K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C DEPLOYMENT         CREATE         - #01-> -1  -1 - 
E PROC               CREATED        - #02->#01 K01 - process.xml->"process" (version:1)
E DEPLOYMENT         CREATED        - #03->#01 K02 - process.xml
E DEPLOYMENT         FULLY_DISTR    - #04->#01 K02 - 
C PROC_INST_CREATION CREATE         - #05-> -1  -1 - new <process "process"> (no vars)
C PROC_INST          ACTIVATE       - #06->#05 K03 - PROCESS "process" in <process "process"[K03]>
E PROC_INST_CREATION CREATED        - #07->#05 K04 - new <process "process"> (no vars)
E PROC_INST          ACTIVATING     - #08->#06 K03 - PROCESS "process" in <process "process"[K03]>
E PROC_INST          ACTIVATED      - #09->#06 K03 - PROCESS "process" in <process "process"[K03]>
C PROC_INST          ACTIVATE       - #10->#06  -1 - START_EVENT "startEv..464f05d" in <process "process"[K03]>
E PROC_INST          ACTIVATING     - #11->#10 K05 - START_EVENT "startEv..464f05d" in <process "process"[K03]>
E PROC_INST          ACTIVATED      - #12->#10 K05 - START_EVENT "startEv..464f05d" in <process "process"[K03]>
C PROC_INST          COMPLETE       - #13->#10 K05 - START_EVENT "startEv..464f05d" in <process "process"[K03]>
E PROC_INST          COMPLETING     - #14->#13 K05 - START_EVENT "startEv..464f05d" in <process "process"[K03]>
E PROC_INST          COMPLETED      - #15->#13 K05 - START_EVENT "startEv..464f05d" in <process "process"[K03]>
E PROC_INST          SEQ_FLOW_TAKEN - #16->#13 K06 - SEQUENCE_FLOW "sequenc..ad498b9" in <process "process"[K03]>
C PROC_INST          ACTIVATE       - #17->#13 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E PROC_INST          ACTIVATING     - #18->#17 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E JOB                CREATED        - #19->#17 K08 - K08 "test" @"task"[K07] 3 retries, in <process "process"[K03]> (no vars)
E PROC_INST          ACTIVATED      - #20->#17 K07 - SERVICE_TASK "task" in <process "process"[K03]>
C JOB                COMPLETE       - #21-> -1 K08 - K08 "test" @""[ -1] -1 retries, in <process ?[?]> (no vars)
E JOB                COMPLETED      - #22->#21 K08 - K08 "test" @"task"[K07] 3 retries, in <process "process"[K03]> (no vars)
C PROC_INST          COMPLETE       - #23->#21 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E PROC_INST          COMPLETING     - #24->#23 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E INCIDENT           CREATED        - #25->#23 K09 - IO_MAPPING_ERROR failed to evaluate expression '{y:x}': no variable found for name 'x',  @"task"[K07] in <process "process"[K03]>
C PROC_INST          CANCEL         - #26-> -1 K03 - UNSPECIFIED "" in <process ?[K03]>
C PROC_INST          TERMINATE      - #27->#26 K03 - PROCESS "process" in <process "process"[K03]>
E PROC_INST          TERMINATING    - #28->#27 K03 - PROCESS "process" in <process "process"[K03]>
C PROC_INST          TERMINATE      - #29->#27 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E PROC_INST          TERMINATING    - #30->#29 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E INCIDENT           RESOLVED       - #31->#29 K09 - IO_MAPPING_ERROR failed to evaluate expression '{y:x}': no variable found for name 'x',  @"task"[K07] in <process "process"[K03]>
E PROC_INST          TERMINATED     - #32->#29 K07 - SERVICE_TASK "task" in <process "process"[K03]>
E PROC_INST          TERMINATED     - #33->#29 K03 - PROCESS "process" in <process "process"[K03]>

-------------- Deployed Processes ----------------------
process.xml ------ 
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:ns0="http://camunda.org/schema/zeebe/1.0" id="definitions_feb15ec1-bd4f-4581-81b2-925711eb4896" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process" isExecutable="true">
    <startEvent id="startEvent_afc2824d-7862-4ade-9315-eb390464f05d">
      <outgoing>sequenceFlow_d74480b7-44b9-4352-9c05-f773bad498b9</outgoing>
    </startEvent>
    <serviceTask id="task" name="task">
      <extensionElements>
        <ns0:taskDefinition type="test"/>
        <ns0:ioMapping>
          <ns0:output source="=x" target="y"/>
        </ns0:ioMapping>
      </extensionElements>
      <incoming>sequenceFlow_d74480b7-44b9-4352-9c05-f773bad498b9</incoming>
      <outgoing>sequenceFlow_626d352b-9dda-45e2-929d-70810c5adb85</outgoing>
    </serviceTask>
    <sequenceFlow id="sequenceFlow_d74480b7-44b9-4352-9c05-f773bad498b9" sourceRef="startEvent_afc2824d-7862-4ade-9315-eb390464f05d" targetRef="task"/>
    <endEvent id="endEvent_192b6c55-07bf-43bf-ab8e-b274e01e0a35">
      <incoming>sequenceFlow_626d352b-9dda-45e2-929d-70810c5adb85</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_626d352b-9dda-45e2-929d-70810c5adb85" sourceRef="task" targetRef="endEvent_192b6c55-07bf-43bf-ab8e-b274e01e0a35"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_8aa8f842-4b04-418f-a4e5-06fab7676134">
    <bpmndi:BPMNPlane bpmnElement="process" id="BPMNPlane_361ddd39-2e26-47cd-9fb6-67681733c806">
      <bpmndi:BPMNShape bpmnElement="startEvent_afc2824d-7862-4ade-9315-eb390464f05d" id="BPMNShape_0302ab37-7f7a-49b3-bb4b-a93938610785">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="task" id="BPMNShape_88a77404-ed5a-4e9e-abc0-aa9711feb0d6">
        <dc:Bounds height="80.0" width="100.0" x="186.0" y="78.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_d74480b7-44b9-4352-9c05-f773bad498b9" id="BPMNEdge_47ef80e5-b38d-4a60-ab44-ed246ccb59bb">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="endEvent_192b6c55-07bf-43bf-ab8e-b274e01e0a35" id="BPMNShape_07ca66c5-30d9-4e54-abfc-a6393ed87dbd">
        <dc:Bounds height="36.0" width="36.0" x="336.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_626d352b-9dda-45e2-929d-70810c5adb85" id="BPMNEdge_c61136cf-9c3d-4dd4-84b0-67258b082121">
        <di:waypoint x="286.0" y="118.0"/>
        <di:waypoint x="336.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>

--------------- Decomposed keys (for debugging) -----------------
 -1 <-> -1
K01 <-> 2251799813685249
K02 <-> 2251799813685250
K03 <-> 2251799813685251
K04 <-> 2251799813685252
K05 <-> 2251799813685253
K06 <-> 2251799813685254
K07 <-> 2251799813685255
K08 <-> 2251799813685256
K09 <-> 2251799813685257

09:49:59.041 [] DEBUG io.camunda.zeebe.util.buffer - Close stream processor
09:49:59.043 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
09:49:59.072 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream-1
09:49:59.073 [stream-1-write-buffer] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
09:49:59.073 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream-1 close 2 readers
09:49:59.074 [] DEBUG io.camunda.zeebe.broker.test - Clean up test files on path /tmp/junit16651370352909439224
09:49:59.074 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
09:49:59.074 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
09:49:59.075 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
09:49:59.075 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully

org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a lambda expression in io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest [Expected all columns to be empty] 
Expecting empty but was: [TEMPORARY_VARIABLE_STORE] within 10 seconds.

	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest.assertThatStateIsEmpty(ProcessExecutionCleanStateTest.java:807)
	at io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest.testTerminatingProcessWithServiceTask(ProcessExecutionCleanStateTest.java:123)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: java.lang.AssertionError: [Expected all columns to be empty] 
Expecting empty but was: [TEMPORARY_VARIABLE_STORE]
	at io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest.lambda$assertThatStateIsEmpty$27(ProcessExecutionCleanStateTest.java:815)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

</p>
</details>

**Environment:**
 - OS: <!-- [e.g. Linux] -->
 - Zeebe Version: `1.0.0-alpha7`
 - Configuration: <!-- [e.g. exporters etc.] -->

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java
@@ -131,7 +131,10 @@ public final class EventAppliers implements EventApplier {
     register(
         ProcessInstanceIntent.ELEMENT_TERMINATED,
         new ProcessInstanceElementTerminatedApplier(
-            elementInstanceState, eventScopeInstanceState, bufferedStartMessageEventStateApplier));
+            elementInstanceState,
+            eventScopeInstanceState,
+            variableState,
+            bufferedStartMessageEventStateApplier));
     register(
         ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN,
         new ProcessInstanceSequenceFlowTakenApplier(elementInstanceState, processState));

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceElementTerminatedApplier.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceElementTerminatedApplier.java
@@ -10,6 +10,7 @@ package io.camunda.zeebe.engine.state.appliers;
 import io.camunda.zeebe.engine.state.TypedEventApplier;
 import io.camunda.zeebe.engine.state.mutable.MutableElementInstanceState;
 import io.camunda.zeebe.engine.state.mutable.MutableEventScopeInstanceState;
+import io.camunda.zeebe.engine.state.mutable.MutableVariableState;
 import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceRecord;
 import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
 
@@ -19,14 +20,17 @@ final class ProcessInstanceElementTerminatedApplier
 
   private final MutableElementInstanceState elementInstanceState;
   private final MutableEventScopeInstanceState eventScopeInstanceState;
+  private final MutableVariableState variableState;
   private final BufferedStartMessageEventStateApplier bufferedStartMessageEventStateApplier;
 
   public ProcessInstanceElementTerminatedApplier(
       final MutableElementInstanceState elementInstanceState,
       final MutableEventScopeInstanceState eventScopeInstanceState,
+      final MutableVariableState variableState,
       final BufferedStartMessageEventStateApplier bufferedStartMessageEventStateApplier) {
     this.elementInstanceState = elementInstanceState;
     this.eventScopeInstanceState = eventScopeInstanceState;
+    this.variableState = variableState;
     this.bufferedStartMessageEventStateApplier = bufferedStartMessageEventStateApplier;
   }
 
@@ -37,5 +41,6 @@ final class ProcessInstanceElementTerminatedApplier
 
     eventScopeInstanceState.deleteInstance(key);
     elementInstanceState.removeInstance(key);
+    variableState.removeTemporaryVariables(key);
   }
 }

--- a/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
@@ -746,6 +746,40 @@ public final class ProcessExecutionCleanStateTest {
     assertThatStateIsEmpty();
   }
 
+  @Test
+  public void testTerminatingProcessWithServiceTask() {
+    // given
+    engineRule
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .serviceTask("task", t -> t.zeebeJobType("test").zeebeOutputExpression("x", "y"))
+                .endEvent()
+                .done())
+        .deploy();
+
+    // when
+    final var processInstanceKey =
+        engineRule.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    engineRule.job().ofInstance(processInstanceKey).withType("test").complete();
+
+    RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .await();
+
+    engineRule.processInstance().withInstanceKey(processInstanceKey).cancel();
+
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_TERMINATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementType(BpmnElementType.PROCESS)
+        .await();
+
+    // then
+    assertThatStateIsEmpty();
+  }
+
   private void assertThatStateIsEmpty() {
     // sometimes the state takes few moments until is is empty
     Awaitility.await()

